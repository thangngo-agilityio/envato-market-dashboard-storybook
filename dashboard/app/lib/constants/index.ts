export * from './routers';
export * from './time';
export * from './images';
export * from './form';
export * from './messages';
export * from './regex';
export * from './texts';
export * from './themes';
export * from './api';
export * from './images';
export * from './menu';
export * from './sidebar';
export * from './charts';
export * from './sidebar';
export * from './common';
export * from './options';
export * from './transaction';
export * from './store';
export * from './status';
export * from './pagination';
export * from './columns';
export * from './faq';
export * from './avatar';
export * from './sizes';
export * from './setting';
export * from './termAndCondition';
export * from './notification';
export * from './authentication';
export * from './discount';
export * from './metadata';
export * from './og-properties';
export * from './chat';
export * from './product';
