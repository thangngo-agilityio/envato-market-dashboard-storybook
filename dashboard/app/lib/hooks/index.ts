export * from './useForm';
export * from './useAuth';
export * from './useStatistical';
export * from './useTransactions';
export * from './useDebounce';
export * from './useSearch';
export * from './usePagination';
export * from './useStore';
export * from './useUser';
export * from './usePinCode';
export * from './useNotification';
export * from './useWallet';
export * from './useMoney';
export * from './useChat';
export * from './useImageUploader';
export * from './useProducts';
export * from './useRecentActivities';
