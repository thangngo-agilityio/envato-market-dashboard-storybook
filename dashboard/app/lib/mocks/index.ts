export * from './spending';
export * from './table';
export * from './transaction';
export * from './product';
export * from './user';
export * from './time';
export * from './statistic';
export * from './chat';
export * from './issues';
export * from './wallet';
export * from './supports';
export * from './recentActivities';
