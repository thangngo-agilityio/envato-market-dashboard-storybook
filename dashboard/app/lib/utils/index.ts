export * from './transaction';
export * from './product';
export * from './time';
export * from './sort';
export * from './user';
export * from './url';
export * from './helpers';
export * from './text';
export * from './localStorage';
export * from './image';
export * from './prefetch';
export * from './toast';
export * from './getErrorMessage';
export * from './firebase';
export * from './recentActivities';
export * from './logAcitvity';
