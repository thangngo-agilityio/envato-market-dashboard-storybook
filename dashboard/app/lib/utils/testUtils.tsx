import { QueryClient, QueryClientProvider } from '@tanstack/react-query';

const queryClient = new QueryClient();

export const renderQueryProviderTest = (component: JSX.Element) => {
  const { render } = testLibReactUtils;
  return render(
    <QueryClientProvider client={queryClient}>{component}</QueryClientProvider>,
  );
};
