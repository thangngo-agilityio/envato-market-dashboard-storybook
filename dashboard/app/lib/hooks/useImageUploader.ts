import { useCallback } from 'react';

// Constants
import { ERROR_MESSAGES, LIMIT_PRODUCT_IMAGES, REGEX } from '@/lib/constants';

// Services
import { uploadImage } from '@/lib/services';

interface ImageUploaderProps {
  onChange: (imagesUpload: string[]) => void;
  onUploadError: (errorMessage: string) => void;
  setPreviewURL: React.Dispatch<React.SetStateAction<string[]>>;
}

export const useImageUploader = ({
  onChange,
  onUploadError,
  setPreviewURL,
}: ImageUploaderProps) => {
  const onDrop = useCallback(
    (acceptedFiles: File[]) => {
      const imagesPreview: React.SetStateAction<string[]> = [];
      const imagesUpload: string[] = [];

      if (acceptedFiles.length > LIMIT_PRODUCT_IMAGES) {
        return onUploadError(ERROR_MESSAGES.UPLOAD_IMAGE_ITEM);
      }

      acceptedFiles.map(async (file) => {
        if (!file) {
          return;
        }

        // Check type of image
        if (!REGEX.IMG.test(file.name)) {
          return onUploadError(ERROR_MESSAGES.UPLOAD_IMAGE);
        }

        // Uploading file
        try {
          onChange([]);
          const previewImage: string = URL.createObjectURL(file);
          const formData = new FormData();

          formData.append('image', file);
          imagesPreview.push(previewImage);

          const result = await uploadImage(formData);
          imagesUpload.push(result);
        } catch (error) {
          onUploadError(ERROR_MESSAGES.UPDATE_FAIL.title);
        }
        setPreviewURL(imagesPreview);
      });

      onChange(imagesUpload);
    },
    [onChange, onUploadError, setPreviewURL],
  );

  return { onDrop };
};

export default useImageUploader;
