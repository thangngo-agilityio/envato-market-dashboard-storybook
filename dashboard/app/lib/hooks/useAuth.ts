import { useRouter } from 'next/navigation';
import { useCallback, useState } from 'react';
import { AxiosError, AxiosResponse } from 'axios';
import { shallow } from 'zustand/shallow';
import { getAuth, signInWithEmailAndPassword } from 'firebase/auth';

// Constants
import {
  END_POINTS,
  ERROR_MESSAGES,
  IMAGES,
  LOGOUT_TIME,
  ROUTES,
} from '@/lib/constants';

// Services
import { MainHttpService, recentActivitiesHttpService } from '@/lib/services';

// Types
import { TUserDetail, EActivity, TActivitiesRequest } from '@/lib/interfaces';

// Utils
import { formatUppercaseFirstLetter, getCurrentTimeSeconds } from '@/lib/utils';

// Stores
import { authStore } from '@/lib/stores';

type TSignUpErrorField = Partial<
  Record<keyof Omit<TUserDetail, 'id' | 'createdAt'>, string>
>;

export type TUserInfo = Omit<TUserDetail, 'password'> | null;

export type TUserAxiosResponse = Omit<
  TUserDetail & { _id: string },
  'id'
> | null;

export type TUseAuth = {
  user: TUserInfo;
  isRemember: boolean;
  date: number;
  setUser: (user: TUserDetail) => void;
  signIn: (
    {
      email,
      password,
    }: {
      email: string;
      password: string;
    },
    isRemember?: boolean,
  ) => Promise<void>;
  signUp: (userInfo: Omit<TUserDetail, 'id' | 'createdAt'>) => Promise<{
    errors?: TSignUpErrorField;
  }>;
  signOut: () => void;
};

export const useAuth = () => {
  const { user } = authStore();
  const [isLogout, setIsLogout] = useState(false);
  const router = useRouter();
  const { updateStore, clearStore } = authStore(
    (state) => ({
      updateStore: state.updateStore,
      clearStore: state.clearStore,
    }),
    shallow,
  );
  const handleSignInWithFirebase = useCallback(
    async (email: string, password: string) => {
      const auth = getAuth();
      return await signInWithEmailAndPassword(auth, email, password).then(
        (userCredential) => {
          const { uid, email } = userCredential.user;

          return {
            uid,
            email,
          };
        },
      );
    },
    [],
  );

  const handleSignIn = useCallback(
    async (
      {
        email,
        password,
        fcmToken = '',
      }: {
        email: string;
        password: string;
        fcmToken?: string;
      },
      isRemember?: boolean,
    ): Promise<void> => {
      try {
        const { uid } = await handleSignInWithFirebase(email, password);

        const { data }: AxiosResponse<TUserAxiosResponse | undefined> =
          await MainHttpService.post<TUserAxiosResponse | undefined>(
            `${END_POINTS.SIGN_IN}`,
            {
              email,
              password,
              fcmToken,
              uid,
            },
            {},
            EActivity.SIGN_IN,
          );

        let localData: TUserInfo | undefined;
        if (data) {
          const { _id, ...rest } = data;
          localData = { ...rest, id: _id };
        }
        return updateStore({
          user: localData,
          isRemember,
          date: getCurrentTimeSeconds(),
        });
      } catch (error) {
        const { response } = error as AxiosError<string>;

        if (!response) {
          throw new Error(
            formatUppercaseFirstLetter(ERROR_MESSAGES.AUTH_INCORRECT),
          );
        }

        throw new Error(formatUppercaseFirstLetter(response?.data));
      }
    },
    [handleSignInWithFirebase, updateStore, user?.id],
  );

  const handleSignUp = useCallback(
    async (
      userInfo: Omit<TUserDetail, 'id' | 'createdAt'> & { fcmToken: string },
    ): Promise<void> => {
      const { email, password, firstName, lastName, fcmToken } = userInfo;
      try {
        const { data }: AxiosResponse<TUserAxiosResponse | undefined> =
          await MainHttpService.post<TUserAxiosResponse | undefined>(
            `${END_POINTS.SIGN_UP}`,
            {
              ...userInfo,
              avatarURL: IMAGES.AVATAR_SIGN_UP.url,
              createdAt: Date.now(),
              email,
              password,
              firstName,
              lastName,
              fcmToken,
            },
            {},
            EActivity.SIGN_UP,
          );

        let localData: TUserInfo | undefined;
        if (data) {
          const { _id, ...rest } = data;
          localData = { ...rest, id: _id };
        }

        return updateStore({ user: localData, date: getCurrentTimeSeconds() });
      } catch (error) {
        const { response } = error as AxiosError<string>;

        throw new Error(formatUppercaseFirstLetter(response?.data));
      }
    },
    [updateStore, user?.id],
  );

  const handleLogout = useCallback(
    async (
      redirectPath?: string,
      option?: keyof Pick<typeof router, 'push' | 'replace'>,
    ) => {
      setIsLogout(true);

      await recentActivitiesHttpService.post<TActivitiesRequest>(
        END_POINTS.RECENT_ACTIVITIES,
        {
          userId: user?.id,
          actionName: EActivity.SIGN_OUT,
        },
      );

      setTimeout(() => {
        clearStore();
        setIsLogout(false);
        router[option ?? 'replace'](redirectPath ?? ROUTES.LOGIN);
      }, LOGOUT_TIME);
    },
    [clearStore, router, user?.id],
  );

  return {
    isLogoutHandling: isLogout,
    setUser: updateStore,
    signIn: handleSignIn,
    signUp: handleSignUp,
    signOut: handleLogout,
    handleSignInWithFirebase,
  };
};
