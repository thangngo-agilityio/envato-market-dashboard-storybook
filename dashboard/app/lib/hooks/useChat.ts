import { Dispatch, MutableRefObject, SetStateAction, useEffect } from 'react';

// Stores
import { authStore } from '@/lib/stores';

// Interface
import { TUserInfo, useGetUserDetails } from '.';
import { TMessages } from '@/lib/interfaces';

// Firebase
import { collection, doc, getDocs, onSnapshot } from 'firebase/firestore';

// Constants
import { AUTHENTICATION_ROLE, FIREBASE_CHAT } from '@/lib/constants';

// Utils
import { db } from '@/lib/utils';
import { MOCK_SUPER_ADMIN } from '../mocks';
import { getAdminDetailsWithId } from '../services';

// TODO: if have real id from firestore
export const useGetRoomChat = () => {
  const user = authStore((state) => state.user);
  const { filterDataUser } = useGetUserDetails(user?.id || '');

  const isAdmin = filterDataUser?.find(
    (user) => user.role === AUTHENTICATION_ROLE.SUPER_ADMIN,
  );

  const adminUserId = isAdmin?._id;
  const comBindIdChat = `${user?.id || ''}${adminUserId}`;
  return comBindIdChat;
};

// Author: Loc Vo
export const getInfoRoomChat = async (user: TUserInfo) => {
  const admin = await getAdminDetailsWithId(user?.id);
  const userUid = user?.uid;

  return {
    roomChatId: `${admin.uid}${userUid}`,
    userId: userUid || '',
    adminId: admin.uid || '',
    avatarUrl: user?.avatarURL || '',
    avatarAdminUrl: MOCK_SUPER_ADMIN?.avatarURL || '',
    displayName: `${user?.firstName} ${user?.lastName} `,
  };
};

export const getLists = async () => {
  const usersCollection = collection(db, FIREBASE_CHAT.USER_CHATS);
  const usersSnapshot = await getDocs(usersCollection);

  const chatList = usersSnapshot.docs
    .map((doc) => ({
      id: doc.id,
      data: doc.data(),
    }))
    .map((item) => Object.values(item.data)[0]) as [];

  return {
    chatList,
  };
};

export const useSubscribeToChat = (
  roomId: string,
  setMessages: Dispatch<SetStateAction<TMessages[]>>,
  boxRef?: MutableRefObject<HTMLDivElement | null>,
) => {
  useEffect(() => {
    if (!roomId) return;

    const unSub = onSnapshot(
      doc(db, FIREBASE_CHAT.CHATS, roomId),
      async (doc) => {
        doc.exists() && setMessages(doc.data().messages);
      },
    );

    if (boxRef?.current) {
      boxRef.current.scrollTop = boxRef.current.scrollHeight;
    }

    return () => {
      unSub();
    };
  }, [setMessages, roomId, boxRef]);
};
