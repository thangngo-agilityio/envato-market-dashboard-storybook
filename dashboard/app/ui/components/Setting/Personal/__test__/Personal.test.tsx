import { screen, waitFor } from '@testing-library/react';

// Components
import UserForm from '..';
import userEvent from '@testing-library/user-event';
import { renderQueryProviderTest } from '@/lib/utils/testUtils';
import { useUpdateUser } from '@/lib/hooks';
import { USER_DETAIL_MOCK } from '@/lib/mocks';

const updateProfileMock = jest.fn();

jest.mock('@/lib/hooks', () => ({
  ...jest.requireActual('@/lib/hooks'),
  useUpdateUser: jest.fn(),
}));

jest.mock('@chakra-ui/react', () => ({
  ...jest.requireActual('@chakra-ui/react'),
  useToast: () => jest.fn(),
}));

describe('Personal Page test cases', () => {
  beforeAll(() => {
    (useUpdateUser as jest.Mock).mockReturnValue({
      data: USER_DETAIL_MOCK,
      useCreateIssues: updateProfileMock,
    });
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  it('renders UserForm with content', () => {
    const { container } = renderQueryProviderTest(<UserForm />);

    expect(container).toMatchSnapshot();
  });

  it('should handle submit form', async () => {
    renderQueryProviderTest(<UserForm />);

    const mockInput = 'mock input';

    const firstNameInput = screen.getByRole<HTMLInputElement>('textbox', {
      name: /first name/i,
      hidden: true,
    });

    await userEvent.type(firstNameInput, mockInput);

    const submitBtn = screen.getByRole<HTMLButtonElement>('button', {
      name: /btn\-save\-profile/i,
      hidden: true,
    });

    await userEvent.click(submitBtn);

    waitFor(() => {
      expect(submitBtn.disabled).toBeTruthy();
    });
  });

  test('handleSubmitForm should be called when is clicked', async () => {
    const { getByTestId } = renderQueryProviderTest(<UserForm />);

    const clickOpenMenu = getByTestId('click-button-update-profile');

    await userEvent.click(clickOpenMenu);

    waitFor(() => {
      expect(updateProfileMock).toHaveBeenCalledWith({
        firstName: 'mock first name',
      });
    });
  });
});
