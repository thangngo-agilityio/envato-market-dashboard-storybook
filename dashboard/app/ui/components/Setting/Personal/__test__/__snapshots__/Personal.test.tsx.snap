// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Personal Page test cases renders UserForm with content 1`] = `
<div>
  <form
    class="chakra-stack css-tl3ftk"
    id="register-form"
  >
    <div
      class="css-9tw58r"
    >
      <section
        class="css-1ujb4zk"
      >
        <h3
          class="chakra-heading css-zyrakf"
        >
          personal information's
        </h3>
        <div
          class="chakra-stack css-pw5to7"
        >
          <div
            class="chakra-form-control css-1kxonj9"
            role="group"
          >
            <label
              class="chakra-form__label css-1rh809e"
              for="field-:r0:"
              id="field-:r0:-label"
            >
              First Name
            </label>
            <div
              class="chakra-input__group css-1091msq"
              data-group="true"
            >
              <input
                class="chakra-input css-1xzh52x"
                id="field-:r0:"
                name="firstName"
                type="text"
                value=""
              />
            </div>
          </div>
          <div
            class="chakra-form-control css-1kxonj9"
            role="group"
          >
            <label
              class="chakra-form__label css-1rh809e"
              for="field-:r1:"
              id="field-:r1:-label"
            >
              Last Name
            </label>
            <div
              class="chakra-input__group css-1091msq"
              data-group="true"
            >
              <input
                class="chakra-input css-1xzh52x"
                id="field-:r1:"
                name="lastName"
                type="text"
                value=""
              />
            </div>
          </div>
        </div>
        <div
          class="chakra-stack css-1vt642t"
        >
          <div
            class="chakra-form-control css-1kxonj9"
            role="group"
          >
            <label
              class="chakra-form__label css-1rh809e"
              for="field-:r2:"
              id="field-:r2:-label"
            >
              Email
            </label>
            <div
              class="chakra-input__group css-1091msq"
              data-group="true"
            >
              <input
                class="chakra-input css-1xzh52x"
                disabled=""
                id="field-:r2:"
                name="email"
                type="text"
                value=""
              />
            </div>
          </div>
          <div
            class="chakra-form-control css-1kxonj9"
            role="group"
          >
            <label
              class="chakra-form__label css-1rh809e"
              for="field-:r3:"
              id="field-:r3:-label"
            >
              Phone Number (optional)
            </label>
            <div
              class="chakra-input__group css-1091msq"
              data-group="true"
            >
              <input
                class="chakra-input css-1xzh52x"
                id="field-:r3:"
                name="phoneNumber"
                type="text"
                value=""
              />
            </div>
          </div>
        </div>
        <h2
          class="chakra-heading css-1tp2z0m"
        >
          Personal Address
        </h2>
        <div
          class="chakra-stack css-1mpdhlb"
        >
          <div
            class="chakra-form-control css-1kxonj9"
            role="group"
          >
            <label
              class="chakra-form__label css-1rh809e"
              for="field-:r4:"
              id="field-:r4:-label"
            >
              Country and Region (optional)
            </label>
            <div
              class="chakra-input__group css-1091msq"
              data-group="true"
            >
              <input
                class="chakra-input css-1xzh52x"
                id="field-:r4:"
                name="country"
                type="text"
                value=""
              />
            </div>
          </div>
          <div
            class="chakra-form-control css-1kxonj9"
            role="group"
          >
            <label
              class="chakra-form__label css-1rh809e"
              for="field-:r5:"
              id="field-:r5:-label"
            >
              City (optional)
            </label>
            <div
              class="chakra-input__group css-1091msq"
              data-group="true"
            >
              <input
                class="chakra-input css-1xzh52x"
                id="field-:r5:"
                name="city"
                type="text"
                value=""
              />
            </div>
          </div>
        </div>
        <div
          class="chakra-stack css-11xraa"
        >
          <div
            class="chakra-form-control css-1kxonj9"
            role="group"
          >
            <label
              class="chakra-form__label css-1rh809e"
              for="field-:r6:"
              id="field-:r6:-label"
            >
              Address (optional)
            </label>
            <div
              class="chakra-input__group css-1091msq"
              data-group="true"
            >
              <input
                class="chakra-input css-1xzh52x"
                id="field-:r6:"
                name="address"
                type="text"
                value=""
              />
            </div>
          </div>
          <div
            class="chakra-form-control css-1kxonj9"
            role="group"
          >
            <label
              class="chakra-form__label css-1rh809e"
              for="field-:r7:"
              id="field-:r7:-label"
            >
              Postal Code (optional)
            </label>
            <div
              class="chakra-input__group css-1091msq"
              data-group="true"
            >
              <input
                class="chakra-input css-1xzh52x"
                id="field-:r7:"
                name="postalCode"
                type="text"
                value=""
              />
            </div>
          </div>
        </div>
        <h2
          class="chakra-heading css-1tp2z0m"
        >
          Social Information
        </h2>
        <div
          class="chakra-stack css-1mpdhlb"
        >
          <div
            class="chakra-form-control css-1kxonj9"
            role="group"
          >
            <label
              class="chakra-form__label css-1rh809e"
              for="field-:r8:"
              id="field-:r8:-label"
            >
              Facebook (optional)
            </label>
            <div
              class="chakra-input__group css-1091msq"
              data-group="true"
            >
              <input
                class="chakra-input css-1xzh52x"
                id="field-:r8:"
                name="facebookURL"
                type="text"
                value=""
              />
            </div>
          </div>
          <div
            class="chakra-form-control css-1kxonj9"
            role="group"
          >
            <label
              class="chakra-form__label css-1rh809e"
              for="field-:r9:"
              id="field-:r9:-label"
            >
              Twitter (optional)
            </label>
            <div
              class="chakra-input__group css-1091msq"
              data-group="true"
            >
              <input
                class="chakra-input css-1xzh52x"
                id="field-:r9:"
                name="twitterURL"
                type="text"
                value=""
              />
            </div>
          </div>
        </div>
        <div
          class="chakra-stack css-11xraa"
        >
          <div
            class="chakra-form-control css-1kxonj9"
            role="group"
          >
            <label
              class="chakra-form__label css-1rh809e"
              for="field-:ra:"
              id="field-:ra:-label"
            >
              Linkedin (optional)
            </label>
            <div
              class="chakra-input__group css-1091msq"
              data-group="true"
            >
              <input
                class="chakra-input css-1xzh52x"
                id="field-:ra:"
                name="linkedinURL"
                type="text"
                value=""
              />
            </div>
          </div>
          <div
            class="chakra-form-control css-1kxonj9"
            role="group"
          >
            <label
              class="chakra-form__label css-1rh809e"
              for="field-:rb:"
              id="field-:rb:-label"
            >
              Youtube (optional)
            </label>
            <div
              class="chakra-input__group css-1091msq"
              data-group="true"
            >
              <input
                class="chakra-input css-1xzh52x"
                id="field-:rb:"
                name="youtubeURL"
                type="text"
                value=""
              />
            </div>
          </div>
        </div>
        <div
          class="css-8jkrjq"
        >
          <p
            class="chakra-text css-rj1t1b"
          />
          <div
            class="css-tz0gqg"
          >
            <button
              aria-label="btn-save-profile"
              class="chakra-button css-1x2j2tm"
              data-testid="click-button-update-profile"
              disabled=""
              form="register-form"
              type="submit"
            >
              Save Profile
            </button>
          </div>
        </div>
      </section>
      <div
        class="css-qymw7k"
      >
        <div
          class="css-0"
        >
          <h4
            class="chakra-heading css-16gtzp5"
          >
            update profile
          </h4>
          <p
            class="chakra-text css-ojirec"
          >
            Profile at least Size
             
            <span
              class="chakra-text css-19sop79"
            >
              300 x 300.
               
            </span>
            Gift to work too.
          </p>
          <div
            class="css-4bwcy0"
          >
            <label
              class="chakra-form__label css-t806qz"
              for="file"
            >
              <img
                alt="This is avatar sign up"
                class="chakra-image__placeholder css-1rb0x8e"
                src="/icons/user.svg"
              />
            </label>
            <div
              class="chakra-input__group css-165prm8"
              data-group="true"
            >
              <div
                class="chakra-input__left-element css-1rkmdgj"
              >
                <label
                  class="chakra-form__label css-g6pte"
                  for="file"
                >
                  <img
                    alt="Edit icon"
                    class="chakra-image css-1nj8bkn"
                    src="/icons/edit.svg"
                  />
                </label>
                <input
                  accept="image/*"
                  class="chakra-input css-ykt8bm"
                  data-testid="upload-image"
                  id="file"
                  type="file"
                  value=""
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </form>
  <span
    hidden=""
    id="__chakra_env"
  />
</div>
`;
